### ISSUE TITLE
NIWC Testbed Update - Slide # 37 - Independent Mariner Selections

### ISSUE THREAD

**Issue by alvarosanuy** [S-101-Portrayal-subWG/Working-Documents#18]
**S-101PT5 Action 24** - Discuss paper S-101PT5-21

[S-101PT5_21_EN_NIWC Test Bed Report_S101PT5_V1.pdf](https://github.com/S-101-Portrayal-subWG/Working-Documents/files/5290033/S-101PT5_21_EN_NIWC.Test.Bed.Report_S101PT5_V1.pdf)

![image](https://user-images.githubusercontent.com/70189016/94640896-2dad4180-0323-11eb-81c2-0d47ef7e519a.png)

**Comment by DavidGrant-NIWC** [comment]
S-101 currently uses three different mechanisms to provide required "Independent Mariner Selections":
1. A context parameter is provided in the portrayal catalogue (e.g. "Full sectors")
1. A viewing group is provided in the portrayal catalogue (e.g. drawing instructions having a Viewing Group of 31031 are associated with "Highlight Document")
1. Nothing is provided - the OEM is responsible for implementation

#22a: In order to support dual-fuel, OEM's should be able to tie a single UI component to both the S-57 and S-101 associated functionality. The S-101 Product Specification should provide implementation guidance.

#22b:
Using context parameters to implement independent selections has performance implications. We recommend removing context parameters where possible, and providing either an independent viewing group or allow the OEM to implement directly.
- Scale min (Turn SCAMIN off): OEM implements directly by ignoring SCAMIN in drawing commands.
- Shallow pattern: Already uses viewing group 23010 - OEM ties to existing function
- Shallow water dangers: Already uses viewing group 24020 (underwater) / 24050 (above water) - OEM ties to existing function
- Unknown (toggle "?" symbol): Already uses viewing group 21010 - OEM ties to existing function
- Update review: OEM implements

For the remaining selections (Full light lines, Simplified symbols, Plain boundaries), either:
1. Continue to provide context parameters
    - OEM must associate to existing S-52 functions
2. Modify the portrayal to always output both sets of drawing instructions using different viewing groups. The OEM could toggle the viewing groups based on the value of the mariner selection.
   - Requires adding / modifying viewing groups, e.g. 27000 is "paper chart buoys" and 27001 is "simplified buoys"
   - Implementation guidance should be added to S-101 PS

**Comment by alvarosanuy** [comment]
Recommend gather views from OEMs.

**Comment by TDYCARHugh** [comment]
Regarding the use of viewing groups to switch between portrayal preferences (paper chart vs simplified etc).
I think I see what Dave is getting at with the viewing groups but I wonder if it solves an implementation problem but makes other things more complex (design of portrayal and user operation).

The original concept of viewing groups was to filter content on/off to manage clutter, regardless of portrayal preferences or settings.  If I understand the proposal it means mixing content filtering and portrayal preferences into one mechanism.
The management of the viewing groups would be more complex because it is adding another level of nesting and a need for mutual exclusive sets of viewing groups.  It could get pretty messy for a mariner to have both simple and complex linestyles or simple and paper chart symbols on at the same time or to go through a long list of viewing groups to switch them all as desired.

In S-100 the use of viewingGroupLayers would help but I don't know how usable it would be for S-52.

**Comment by HolgerBothien** [comment]
In my opinion viewing groups can be used for the independent mariner selections when there is already a viewing group defined for it. A problem is that viewing groups are also the base for display categories and viewing group layers. The viewing groups for the independent mariner settings must be taken out of the groups that defines categories and layers to make them 'independent'.
Another problem is that the 'independent mariner settings are actually subdivided in different groups.
1. Tick box settings (on/off) to be used during portrayal
2. Selection of values (paper chart/simplified) and (plain/symbolized) - needs a radio or combo box in the interface
 actually that are selection of lookup tables (rules).
3. Settings that only needs consideration during the rendering (SCAMIN)
4. Settings that needs multiple values - date depended needs a mode, depending on that mode a start and/or end date.

**Comment by HannuPeiponen** [comment]
Dual-fuel (DF) use as whole is not yet defined in details - defined is just the target that DF ECDIS can use both S-57 and S-101. Open is the style, for example, alternatively (i.e. one at same time, but not both), adjacent (i.e. separate coverage areas) or mixed (i.e. free mixing of both in the scale stack, zoom in or out in same location may cause presentation of either).
Further interoperability between S-57 and S-101 in a DF ECDIS is not yet defined in details. This interoperability could be based, for example, on techically integrated presentation rules (i.e. looks like this style is behind this Mariner Selections issue from David Grant) or on technically separate presentation rules (i.e. leave S-52 as it has been).
I have interpreted comments of David, Hugh and Holger so that "integrated"/"associated" operation of mariner selections for S-57 and S-101 ENC charts would require amending the current S-52 for compliance with the common integrated control of the presentation.
One should note also that theoretically an S-100 engine based on plug&play concept do not understand anything about the meaning (i.e. for what purpose) of each context parameter. For S-100 engine each context parameter is just a selector with a machine readable sting of characters to be presented for the user. The S-52 implementation is quite opposite, the manufacturer has created the human machine interface based on the manufacturer's own interpretation of the printed document.
The portrayal of S-101 has off course a lot in common with presentation of S-52. But S-101 is intentionally different as there are changes or "improvements" over the functionality of S-52. Integration of Mariner selectors to be totally common between these two would mean either to abandon all "improvements" of S-101 or amending S-52 to require same change/improved functionality as S-101.
The common integrated Mariner selection control would also mean that the S-101 implementation would no more be 100% machine-reading. The S-52 is not machine-readable for the Mariner selectors. Common Mariner selectors would mean that the manufacturer's implementation of DF ECDIS should integrate the S-52 printed rules with S-101 machine readable context parameters, viewing groups, etc. The result would be "frozen" portrayal of S-101 as any change to the portrayal would cause a need to amend the printed rules based integration with S-52.
My opinion is that the implementation of the DF ECDIS should not mean new edition of S-52 in order to make the S-52 "integrateable" with S-101. Simply the S-52 should remain as it has been.

**Comment by DavidGrant-NIWC** [comment]
# TL;DR:
Mostly OBE - two recommended actions:

- [ ] Action 1: Recommend Alvaro to coordinate assignment/registration of viewing groups for independent selectors listed below.

|Selector|Selector VG|Affected drawing instructions/VGs|
|:---:|:---:|:---:|
|Accuracy|?|31010, 31011|
|Highlight date dependent|?|SY(CHDATD01) (currently has no assigned viewing group)|
|Highlight info|?|31030|
|Highlight document|?|31031|
|Shallow pattern|?|DIAMOND01 area fill (needs a viewing group independent of the depth area which is not part of the base display)|
|Shallow water dangers|?|24020, 24050|
|Unknown|?|21010|
|Update review|?|SY(CHRVDELN) etc. (currently has no assigned viewing group)|
|Contour label|?|33021, 33022|

- [ ] Action 2: Recommend update S-98 to include requirement and guidance for Date dependent independent mariner selector.

|Selector|Function|Recommended impementation|
|:---:|:---:|:---:|
|Date dependent|To turn on and off the display of temporal objects by viewing a date range|OEM responsibility, requirement should be added to the S-101 PS. Relates to: ![image](https://user-images.githubusercontent.com/62906211/129602852-1ddf538e-1d8b-4e56-9753-6c6935406eb2.png)<br/><br/>_Note 1: this is not to turn on/off CHDATD01._<br/><br/>_Note 2: The description of this function is not worded very clearly. I assume enabling activates the mariner input date, and disabling activates the current date. Disabling is equivalent to the mariner resetting or selecting "Today" on the date input UI function, so I am not sure if this selector is really needed._|

# Long version:
I think this is mostly OBE assuming upcoming proposals are accepted. The following independent mariner selections can leverage the proposed support for text groups (described below): Accuracy, Highlight date dependent,  Highlight info, Highlight document,  Shallow pattern, Shallow water dangers, Unknown, Update review, Contour label.

Independent mariner selections which affect viewing groups could be implemented similar to text groups (which also need to be 'independent'). See [Proposal: multiple viewing groups per drawing instruction](https://github.com/IHO-S100WG/TSM8/issues/9). Note that S-100 has no equivalent concept to S-52 "text groups".

With the proposed change, the rendering of a drawing instruction requires all viewing groups to be enabled. This facilitates implementing independent mariner selections which rely on viewing groups, to include text groups. e.g.:

![image](https://user-images.githubusercontent.com/62906211/129595579-46afa352-aca0-4a7c-9d42-367596e2325f.png)

In order to implement these viewing group based selectors additional viewing groups need to be assigned for the selectors:
|Selector|Selector VG|Affected drawing instructions/VGs|
|:---:|:---:|:---:|
|Accuracy|?|31010, 31011|
|Highlight date dependent|?|SY(CHDATD01) (currently has no assigned viewing group)|
|Highlight info|?|31030|
|Highlight document|?|31031|
|Shallow pattern|?|DIAMOND01 area fill (needs a viewing group independent of the depth area which is not part of the base display)|
|Shallow water dangers|?|24020, 24050|
|Unknown|?|21010|
|Update review|?|SY(CHRVDELN) etc. (currently has no assigned viewing group)|
|Contour label|?|33021, 33022|

For the remaining selectors the original issue was proposing:
![image](https://user-images.githubusercontent.com/62906211/129605515-1f546849-9b44-4f21-bac6-aef389e15098.png)
The proposed changes to Shallow pattern and Shallow water dangers are included above. The proposed change for Scale min can be withdrawn. Other proposed optimizations (e.g. output multiple portrayals for a given feature and switch between them, turn off SCAMIN during rendering rather than portrayal) are also withdrawn; optimizations can be considered later if needed/desired.

|Selector|Function|Recommended impementation|
|:---:|:---:|:---:|
|Date dependent|To turn on and off the display of temporal objects by viewing a date range|OEM responsibility, requirement should be added to the S-101 PS. Relates to: ![image](https://user-images.githubusercontent.com/62906211/129602852-1ddf538e-1d8b-4e56-9753-6c6935406eb2.png)<br/><br/>_Note 1: this is not to turn on/off CHDATD01._<br/><br/>_Note 2: The description of this function is not worded very clearly. I assume enabling activates the mariner input date, and disabling activates the current date. Disabling is equivalent to the mariner resetting or selecting "Today" on the date input UI function, so I am not sure this is really needed._|
|Full light lines|Full light sector lines|Context parameter|
|Paper chart/Simplified symbols|Point symbol style|Context parameter|
|Plain/Symbolized boundaries|Line symbol style|Context parameter|
|Scale min|Turn SCAMIN Off|Context parameter|
|Four shades|Four Colour Depth Shades|Context parameter|
|National language|3 char language code|Context parameter|

**Comment by DavidGrant-NIWC** [comment]
>HannuPeiponen commented
Dual-fuel (DF) use as whole is not yet defined in details [...] This interoperability could be based, for example, on technically integrated presentation rules (i.e. looks like this style is behind this Mariner Selections issue from David Grant) or on technically separate presentation rules (i.e. leave S-52 as it has been).
> I have interpreted comments of David, Hugh and Holger so that "integrated"/"associated" operation of mariner selections for S-57 and S-101 ENC charts would require amending the current S-52 for compliance with the common integrated control of the presentation.

I think common / integrated control should be the goal to avoid confusing the mariner by presenting two sets of configuration items which are not synchronized. E.g. shallow pattern on in S-57 and off in S-101. This could potentially be hazardous (depends on the details of DF implementation). If technically separate presentation rules are used then this should be taken into consideration when defining DF operation, and it should be very clear to the mariner that S-101 and S-57 are separate / distinct, are configured independently, and have independent presentation.

Also, I am not advocating changing S-52 (except for perhaps adding viewing groups for a few existing symbols which don't have an assigned viewing group). Rather, I believe the OEM should tie S-101 functionality to existing user interface components where possible. In some cases this is straightforward (e.g. matching viewing groups). However, it becomes challenging where functionality has changed, e.g. National language, or the proposed change to Shallow pattern.

> [...] an S-100 engine based on plug&play concept [doesn't know the purpose] of each context parameter. [...] The S-52 implementation is quite opposite [...] The common integrated Mariner selection control would also mean that the S-101 implementation would no more be 100% machine-reading.
>  [...] Common Mariner selectors would mean that the manufacturer's implementation of DF ECDIS should integrate the S-52 printed rules with S-101 machine readable context parameters, viewing groups, etc. The result would be "frozen" portrayal of S-101 as any change to the portrayal would cause a need to amend the printed rules based integration with S-52.

Agree. The linkage to S-52 would likely need to be hard-coded based on information provided in the PS or S-98. This does not greatly affect machine-reading of S-101 - selectors, context parameters and rules can still be added/updated/removed. Bugs can be fixed. New features can be added. Only those aspects tied to DF operation would be frozen. Once the DF transition is complete the "frozen" components can be updated as desired.

This is analogous to other items which are necessarily hard-coded, such as the S-101 "Radar Overlay" context parameter. It is assumed the OEM will hard-code this context parameter to the RADAR function of their software, rather than have the mariner manually toggle the setting.

> My opinion is that the implementation of the DF ECDIS should not mean new edition of S-52 in order to make the S-52 "integrateable" with S-101. Simply the S-52 should remain as it has been.

See above; I agree that S-52 should not change.

**Comment by alvarosanuy** [comment]
**Decisions and Actions at 01DEC2021 (post 2nd meeting in November 2021):**

![image](https://user-images.githubusercontent.com/70189016/144342328-672771dc-0a2b-4c38-80ca-8948a029a69b.png)

**Comment by alvarosanuy** [comment]
**Decision made at Portrayal subWG meeting on 26/7/22**

1. Leave this issue open as no progress has been made..
2. **NIWC** to organise and chair a focus group meeting as directed by S-101PT decision **PortSG-50**.
3. No impact on publication of PC 1.0.2 or 1.1.0.

**Comment by alvarosanuy** [comment]
**Portrayal subWG meeting - 12th January 2023**

1. No progress on this task
2. OEMs are encouraged to commence implementation and testing and provide feedback to NIWC ASAP.

**Comment by DavidGrant-NIWC** [comment]
See also #119

This is a list of mariner selectors which are required by S-52 PL 10.3.4.4 but which do not alter the portrayal output, and therefore are not implemented using portrayal context parameters. Per the discussion above, all except for "Date dependent" will be implemented using "independent" viewing group layers (a viewing group layer which is not included in any display mode such as "Base", "Standard", or "All Other").

Implementation:
* Viewing groups and group layers needed for implementation will be named based on the selector name. For instance, "Shallow pattern" will use "ShallowPatternVG" and "ShallowPatternVGL".
* Add new viewing groups to the portrayal catalog. For instance, "ShallowPatternVG", "HighlightDateDependentVG", etc.
* Add new viewing group layers to the portrayal catalog. For instance, "ShallowPatternVGL", "HighlightDateDependentVGL", etc.
* Update portrayal rules so that drawing instructions are associated with multiple viewing groups. For instance, the rule which generates the diamond fill pattern (shallow water pattern) currently references viewing group 23010. It will be updated to reference both 23010 and "ShallowPatternVG".

If the implementation is deemed acceptable, then the viewing groups and viewing group layers used should be registered.

Note that OEM's will need to initialize the default value since S-100 doesn't support specifying the default value of a viewing group. This requirement could be eliminated via S-100 change proposal if desired.

Default values are from IEC 61174 Table 3.
|Selector|Default Value|Description|Notes|
|:---|:---|:---|:---|
|Accuracy|false|Identify low accuracy data via symbol LOWACC01, viewing group 31011|Viewing Group Layer|
|Date dependent|Current date|Date Dependent Objects – to toggle the display of temporal objects by viewing a date range.|OEM responsibility. Sets the date / date range. OEM uses value to filter portrayal drawing instructions during rendering / pick report. ![image](https://user-images.githubusercontent.com/62906211/216163090-e2e3b4db-a03b-449e-8a79-d35a3326bde1.png)|
|Highlight date dependent|Off|Indication of date dependent objects – to turn on and off the display of symbol CHDATD01|Viewing Group Layer|
|Highlight info|Off|Additional Information - viewing group 31030 (INFORM, NINFOM)|Viewing Group Layer|
|Highlight document|Off|Additional Documents - viewing group 31031 (TXTDSC, NTXDS, PICREP)|Viewing Group Layer|
|Shallow pattern|false|Show a diamond pattern over shallow water|Viewing Group Layer. Updated S-52 description. Mandatory despite note in PL 13.1.5.|
|Unknown|On|Unknown Objects - to turn on the display of objects which are not specified in S-52 standard – viewing group 21010|Viewing Group Layer. Why would you ever want this disabled (seems unsafe)?|
|Update review|Off|Review of Updates –This function turns on colour highlighting for the objects which have undergone modification in the process of the latest accepted correction|Viewing Group Layer / TBD.|
|Contour label|Off|Contour Labels - to turn on the display of contour labels (incl. label of safety contour) - viewing group 33021, 33022|Viewing Group Layer. Optional.|

**Comment by DavidGrant-NIWC** [comment]
@TDYCARHugh @HolgerBothien @HannuPeiponen 

For the independent mariner selections listed above (ignoring Date dependent), can you please clarify the intended implementation:
* [ ] An additional "switch" which, in conjunction with the viewing group, controls the object visibility
  * both must be enabled for the object to be visible
  * For instance, the diamond pattern over shallow water is:
    * ON: (display mode = "Standard" or "All Other") AND ("23010" = on) AND ("Shallow pattern" = on)
    * OFF: (display mode = "Base") OR ("23010" = off) OR ("Shallow Pattern" = off)
* [ ] The equivalent of a viewing group layer
  * the object visibility is toggled with the selector (the selector toggles the viewing group, and vice-versa)
  * For instance, the diamond pattern over shallow water is:
    * ON: (display mode = "Standard" or "All Other") AND ("Shallow pattern" = on)
    * OFF: (display mode = "Base") OR ("Shallow Pattern" = off)

As written in S-52 it seems that the first method is required, but I suspect this would be confusing for the user (three user interface elements need to be coordinated to display the object). The second method seems preferable (two user interface elements control the object).

This is further confused by the fact that the viewing group associated with most of these is also included in some other viewing group layer.
* The shallow water overlay works with viewing group "23010", which is included in viewing group layer "10 - Miscellaneous (Standard)".
* "Accuracy" works with viewing groups "31010" and "31011", which are included in viewing group layer "18 - Miscellaneous (Other)".

**Comment by HolgerBothien** [comment]
As I see it, none of the two options is what the user expects.
The user expects a single control that switches e.g. the shallow water pattern independently of any other control.
That's why it is called independent mariner selection (IMS)
If the IMS is defined by a viewing group, this viewing group must be excluded from the display modes and viewing group layers.

**Comment by DavidGrant-NIWC** [comment]
> If the IMS is defined by a viewing group, this viewing group must be excluded from the display modes and viewing group layers.

We could implement them this way (equivalent to S-52 Text Groups / Text Group Layers) and I agree that this would be the most "user friendly" implementation, however this does not agree with the current S-52 requirements.

For instance, the "Shallow pattern" selector should toggle the "Shallow Water Pattern". The pattern has a display category (Standard), and the assigned viewing group (23010) is included in a viewing group layer (10 - Miscellaneous).

![image](https://user-images.githubusercontent.com/62906211/217584059-b2ce61ca-1dea-4098-a746-e779317d7743.png)

![image](https://user-images.githubusercontent.com/62906211/217585130-0a4100b4-f665-4ff2-8b08-97d626218190.png)

* [ ] Should the S-101 implementation change the existing requirements by removing the display category and viewing group layer assignments?
  * [ ] Should we also propose these changes to the ENCWG?

### S-52 Requirements

|Selector|Display Element(s)|Display Category|Viewing Group Layer|Viewing Group(s)|
|:---|:---|:---|:---|:---
|Accuracy|CATZOC pattern ![image](https://user-images.githubusercontent.com/62906211/217600078-5dd8028b-6e2b-4961-8f42-021c6eda9ebb.png), LOWACC01 ![image](https://user-images.githubusercontent.com/62906211/217599639-57503159-64d3-45fa-8bb9-3916c3a348cd.png)|All Other|18 - Miscellaneous (Other)|31010, 31011|
|Highlight date dependent|CHDATD01 ![image](https://user-images.githubusercontent.com/62906211/217600407-14cd6246-fcda-46e0-af11-d17bad932dae.png)|All Other|18 - Miscellaneous (Other)|31032|
|Highlight info|INFORM01 ![image](https://user-images.githubusercontent.com/62906211/217600651-d3e60527-ecd7-4d77-a32d-abfdce7734d6.png)|All Other|18 - Miscellaneous (Other)|31030|
|Highlight document|INFORM01 ![image](https://user-images.githubusercontent.com/62906211/217600677-b17d0c85-b23d-4d2d-960c-9c2f7935c5c1.png)|All Other|18 - Miscellaneous (Other)|31031|
|Shallow pattern|DIAMOND1 ![image](https://user-images.githubusercontent.com/62906211/217602845-d2cead13-c9dc-4807-aacd-56fb94955fdd.png)|Standard|10 - Miscellaneous (Standard)|23010|
|Unknown|QUESMRK1 ![image](https://user-images.githubusercontent.com/62906211/217601138-6ab687c2-9c6a-49dc-a69d-d2b12fd60e53.png)|Standard|3.1 Buoys, beacons, structures _[this seems like an error?]_|21010|
|Update review|CHCRID/CHCRDEL/etc. ![image](https://user-images.githubusercontent.com/62906211/217602345-78ffc725-a8eb-4f10-a963-612644e7ffd1.png)|Inherited from updated object|Inherited|Inherited|
|Contour label|SAFCONnn ![image](https://user-images.githubusercontent.com/62906211/217601391-b2b33dd9-42d9-4e9e-95a6-09ed7bf85a72.png)|All Other|18 - Miscellaneous (Other)|33021, 33022

**Comment by TDYCARHugh** [comment]
I remember in early ECDIS trials the shallow water pattern was created to resolve a problem at night when the entire screen was covered in one colour it was hard to know which colour it was.  If we could make a rule that always showed the shallow pattern when using the night colours perhaps the single on/off would not really be needed and the shallow water pattern could just be an optional viewing group when using the brighter colours.

**Comment by DavidGrant-NIWC** [comment]
`National language` will likely be removed from the context parameters and implemented as an independent mariner selection. This matches the S-52 requirements, better supports dual-fuel, and is more user friendly for the mariner.

See #104 and #119
Awaiting result of [DCEG #60](https://github.com/iho-ohi/S-101-Documentation-and-FC/issues/60)

**Comment by DavidGrant-NIWC** [comment]
https://user-images.githubusercontent.com/62906211/224846875-fc23c008-34ba-4375-91b8-b9e48c1d08eb.mp4

The independent mariner selections are now implemented in the 1.1.0 portrayal catalog. I don't really like the way it works, but it agrees with the S-52 requirements. I suspect what is written in S-52 doesn't match what was intended.

I suggest waiting to register the new viewing groups and viewing group layers until we resolve the following:
* [ ] Should the S-101 implementation change the existing requirements by removing the display category and viewing group layer assignments?
  * [ ] Should we also propose these changes to the ENCWG?

If there is no feedback, we can register the new VG's/VGL's and close the issue as completed.

**Comment by alvarosanuy** [comment]
I'm happy to delay the registration of VG and VGL until the preferred implementation of IMS is further discussed by the PsWG.

**Based on NIWCs comment PC 1.1.0 implements IMS as per S-52. This is the goal for PC 1.1.0.**
Having said that, it seems that S-52 instructions do not provide the most user friendly experience to mariners and the topic deserves further discussion. 

I propose we keep this issue open (currently labelled as 'PC 1.2.0 or later'), continue adding our views on this issue and aim at reaching some agreement on the best way forward at the next PsWG meeting which would be scheduled for May 2023.

**Comment by alvarosanuy** [comment]
**Decisions made at Portrayal subWG meeting on 10/5/23**

- [ ] **Alvaro** to seek S101PT advice (at S101PT10) on the feasibility of deviating from current S-52/S-98 direction when implementing IMS in order to provide a functionality that is more 'user friendly'. For the record, it seems that a number of OEMs are already implementing IMS differently to what is strictly documented in the IHO instruments. @DavidGrant-NIWC  - Is there any S-64 test for this?
- [ ] Independent of the outcome, a **new S-164 tes**t should be developed to ensure that during the DF-ECDIS period, S-52 and S-101 performance is consistent and delivers the same practical experience to the end user.
- [ ] If approved to proceed with amendments to S-98 - **NIWC** to prepare and submit the corresponding S-98 change proposal.

**Comment by DavidGrant-NIWC** [comment]
> * For the record, it seems that a number of OEMs are already implementing IMS differently to what is strictly documented in the IHO instruments. @DavidGrant-NIWC  - Is there any S-64 test for this?

S-64 section 3.3 is labeled "Independent Mariner Selections" but also checks other settings, such as context parameters. It doesn't check all of the IMS. The checks which are present are mostly done in `Other` or involve text groups (text groups aren't included in any display mode).

**Comment by alvarosanuy** [comment]
@DavidGrant-NIWC - I noticed that the behavior (in Base Display) of the 'Shallow water pattern' in your video above does not agree with the image shown in S-64 test **3.3.6 Shallow pattern**. 

After some reading, it looks like the tables mapping VG/VGL to Display Modes in S-52 and S-98, are only to determine the expected display content when each of the display modes are selected. In the case of 'Shallow water pattern', it seems that, by default, it has to be shown in Standard display but, as an IMS, it should be possible to turn it OFF (the Standard display would then be downgrade to Base+). Similarly, and I think S-64 test 3.3.6 tries to demonstrate this, the shallow water patter can be added to Base display. This behavior makes the Shallow Water pattern option an IMS. The mappings of VG/VGL against Display Modes would be to structure and drive the minimum content for each DM only.
If OEMs are going to expose every single VG to mariners, then all of them become IMS.
Maybe the original S-52 expectation was that OEMs would only expose the VGL plus a limited number of VG which would become IMS (i.e. 23010 'Shallow Water pattern')..... My point is, if soundings can be turned On/Off by mariners at any moment, shouldn't they be called an IMS??

**Comment by DavidGrant-NIWC** [comment]
Note the order of the selections in test 3.3.6. The pattern is not added to `Base`, it is selected prior to selecting `Base`.

In any case, I agree that your interpretation of how to implement display modes is one possibility, but in my opinion S-52 is unclear on exactly how these should be implemented so there are multiple possible interpretations. It can be assumed that multiple methods are in use in currently fielded systems.

Additionally, S-100 imposes restrictions on the implementation due to it's machine readability. Whereas in S-52 special processing could be used for implementing functionality such as the shallow water pattern selector, that should not be the case in S-101.

Finally, the process of changing display modes / vgls / vgs becomes very complicated when you start considering multiple versions of multiple catalogues from multiple products (which is why we plan to switch our testbed implementation).

### Some possible display mode implementations
1. Changing a display mode toggles the VG/VGL's to match the selected mode. This is what we do in our testbed, but we plan on switching to option (2) at some point.
    * Selecting `Base` turns off everything in `Standard` and `Other`(both on the screen and in the UI).
    * Changing a VGL changes the Display Mode and the VG's.
    * Changing a VG changes the VGL(s) and Display Mode.
1. Selecting a display mode enables the user interface selection of VGLs/VGs which are in the selected mode. VGLs/VGs not in the selected mode are disabled (grayed out, but retain their current setting).
    * Selecting `Base` turns off everything in `Standard` and `Other` (only on the screen, UI selections are unchanged)
    * Changing the display mode doesn't change the VGL/VG settings
    * Changing a VGL/VG does not change the display mode.
1. Use either of the previous methods, but also modify the settings to match the IMS whenever it is changed. Note that for some implementations this could require changing a different IMS setting if the display mode or a shared VGL is modified.

**Comment by alvarosanuy** [comment]
**Below is a copy of an email sent by Hannu after S101PT10**


You have got this email based on S-101PT meeting today. In your presentation you requested total independency from the display category (base/standard/other) of the related object. My opinion is that kind of decision will have unwanted consequentials. Below is the related slide from your presentation:
![image](https://github.com/S-101-Portrayal-subWG/Working-Documents/assets/70189016/383e5d26-134c-410a-9aa4-9b7d4cd87a81)
</span><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;">S</span></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span lang="EN-US" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;"> </span></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;">------------------------------------------------</span></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><b><span lang="EN-GB" style="border: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 12pt; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;">IHO S-52 Preslib specification on independent mariner selectors</span></b></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;">For your information below is the IHO S-52 rule:</span></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;"> </span></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt 36pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><b><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;">10.3.4.4 Independent Mariner Selections</span></b></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt 36pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;">There are a number of settings in ECDIS that can be operated by the Mariner that will alter the charted display.<span> </span><b>Each of these objects has a display category assigned<span> </span><span style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: red !important;">but must only be shown in that display category<span> </span></span>if the Mariner has decided to select this option</b>. Listed alphabetically below are the Mariner selectors that must be available in ECDIS, the name of the selector in ECDIS, and a functional description. Three optional Mariner selectors, shown at the bottom of the table, may also be included in ECDIS.</span></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt 36pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;"> </span></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;">As you see the precondition for display is that mariner has selected the related display category. Then within the display category the “independent mariner selector” change the presentation as specified for each case.</span></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;"> </span></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;">Below is the table 10.3.4.4. I added a comment column by me to inform what is happening (by reading these you can understand the unwanted consequences).</span></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;"> </span></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt 72pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;"> </span></p><div class="R1UVb" style="border: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-variant-alternates: inherit; font-weight: 400; font-stretch: inherit; font-size: 15px; line-height: inherit; font-family: &quot;Segoe UI&quot;, &quot;Segoe UI Web (West European)&quot;, &quot;Segoe UI&quot;, -apple-system, BlinkMacSystemFont, Roboto, &quot;Helvetica Neue&quot;, sans-serif; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: rgb(36, 36, 36); overflow: hidden; position: relative; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; height: 3021px; width: 594.125px;"><div class="qF8_5" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit; opacity: 0; position: absolute; z-index: -1;"><button type="button" class="ms-Button ms-Button--icon wD8TJ root-479" title="Show original size" aria-label="Show original size" data-is-focusable="true" style="border: none; font-style: inherit; font-variant: inherit; font-weight: 400; font-stretch: inherit; font-size: 16px; line-height: inherit; font-family: controlIcons !important; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; margin: 0px; padding: 8px; vertical-align: baseline; color: var(--white); outline: transparent; position: absolute; -webkit-font-smoothing: antialiased; border-radius: 2px; box-sizing: border-box; cursor: pointer; display: inline-block; text-decoration: none; text-align: center; width: 32px; height: 32px; background: var(--neutralPrimary); user-select: none; box-shadow: 0 0 5px var(--neutralSecondary); left: 5px; top: 5px; z-index: 1;"><span class="ms-Button-flexContainer flexContainer-164" data-automationid="splitbuttonprimary" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit; display: flex; height: 16px; flex-wrap: nowrap; justify-content: center; align-items: center;"><i data-icon-name="FullScreen" aria-hidden="true" class="ms-Icon root-90 css-297 ms-Button-icon icon-263" style="display: inline-block; font-size: 16px; margin: 0px 4px; height: 16px; line-height: 16px; text-align: center; flex-shrink: 0; -webkit-font-smoothing: antialiased; font-style: normal; font-weight: normal; speak: none; font-family: controlIcons;"></i></span></button></div>

Clause | ECDIS Implementation | Name of Selector in ECDIS | Function Description | Peiponen comment on implementation
-- | -- | -- | -- | --
14.2 | Mandatory | Accuracy | Independent pattern selection of CATZOC, viewing group 31010. Symbol LOWACC01, viewing group 31011 | This is possible when “other” has been selected as display category
10.4.1 | Mandatory | Date dependent | Date Dependent Objects – to turn on and off the display of temporal objects by viewing a date range. | In general, this is applied in any object which is selected to display by its display categoryNote that for example, when display base is selected no object from standard or other are suddenly displayed. Still only object from base with date dependent attributes are treated for the date range
13.2.7 | Mandatory | Full light lines | Full Light Sector Lines | Lights are in display category standard. When base is selected then there are no lights (and because of no lights no extension of their sectors)
10.4.1.1 | Mandatory | Highlight date dependent | Indication of date dependent objects – to turn on and off thedisplay of symbolCHDATD01 | See date dependent comment above
10.6.1.1 | Mandatory | Highlight infoHighlight document | Additional Information - viewing group 31030 (INFORM, NINFOM)Additional Documents – viewing group 31031 (TXTDSC, NTXDS, PICREP) | This is possible when “other” has been selected as display category
12 | Mandatory | Paper chart / simplified symbols | Point symbol style | Selection between two drawing styles. By nature independent of base/standard/other
12 | Mandatory | Plain / Symbolized boundaries | Line symbol style | Selection between two drawing styles. By nature independent of base/standard/other
10.4.2 | Mandatory | Scale min | Turn SCAMIN Off | In general, this is applied in any object which is selected to display by its display category.Note that for example, when display base is selected no object from standard or other are suddenly displayed. Still only object from base are treated for the SCAMIN
10.5.7 | Mandatory | Shallow pattern | Shallow Water Pattern | This is not connected to viewing groups of charted objects, see my explanation on S-52 clause 10.3.4.6 below
13.2.19 | Mandatory | Shallow water dangers | Isolated Dangers in Shallow Water – to turn on the display of isolated danger objects which are located in the unsafe waters - viewing group 24020, 24050 | This is for possible isolated dangers which lay in unsafe water based on user selected safety contour. This selector change them from non- dangerous symbol to “isolated danger” symbol type.
10.3.3.4 | Mandatory | Unknown | Unknown Objects - to turn on the display of objects which are not specified in S-52 standard – viewing group 21010 | Default for display of unknown is not to display at all. This enable them to be displayed when “standard” or “other” is selected
10.7.1.5 | Mandatory | Update review | Review of Updates – This function turns on colour highlighting for the objects which have undergone modification in the process of the latest accepted correction; | In general, this is applied in any object which is selected to display by its display category.
14.2 | Optional | Contour label | Contour Labels – to turn on the display of contour labels (incl. label of safety contour) – viewing group 33021, 33022 | This is possible only when “other” has been selected
13.2.14 | Optional | Four shades | Four Colour Depth Shades | This is independent of display category as all depth areas are in base.
10.6.1.2 | Optional | National language | National Language NOBJNM, text group 31 | In general, this is applied in any object which is selected to display by its display category.


The shallow water pattern is a specific case: It is an object independent like north arrow or scale bar. And it has its own detailed implementation rule, see below clauses 10.3.4.6 and 10.5.7. Selection of display category “other” include all from “base” and “standard”. Therefore  the shallow water pattern is only missed when “base” is selected.  Technically this could be done by changing the associated viewing group (currently 23010), for example to 13010.</span></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;"> </span></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt 36pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><b><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;">10.3.4.6 Viewing groups</span></b></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt 36pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;">The revised IMO Performance Standards for ECDIS, section 5.5. state, "It should be easy [for the mariner] to add or remove information from the ECDIS display," although, "It should not be possible to remove information in the Display Base."</span></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt 36pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;">The viewing groups in table 14.2 are a framework on which the ECDIS manufacturer can base his own method of providing this capability. The minimum mandatory ECDIS implementation is described in 14.3 and 14.5.</span></p><p class="x_MsoNormal" style="margin: 0cm 0cm 0.0001pt 36pt; font-size: 11pt; font-family: Calibri, sans-serif; color: rgb(36, 36, 36); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span lang="EN-GB" style="border: 0px; font: inherit; margin: 0px; padding: 0px; vertical-align: baseline; color: inherit;">Viewing groups are 'on' or 'off' switches for use by the Mariner to control the information appearing on the display.<span> </span><b>An item in the viewing group table may be</b><span> </span>a chart object; a Mariners' or other time-variable object; a special symbol such as the "depth less than safety contour" pattern; or<span> </span><b>a non-ENC feature such as the shallow water pattern</b>. In Presentation Library Edition 3.3 further 'symbol viewing groups' were added, to allow auxiliary symbols such as contour labels, the 'low accuracy' symbol, etc., to be switched on or off without affecting the primary symbolisation of the object.</span></p>
![image](https://github.com/S-101-Portrayal-subWG/Working-Documents/assets/70189016/c0306996-bf9f-4991-a743-50ed2dc337f9)

#________________________________________________________<br aria-hidden="true">Hannu Peiponen</span></p>

**Comment by alvarosanuy** [comment]
**Response by @DavidGrant-NIWC to @HannuPeiponen email above.**

Hi Hannu, 

Thanks for confirming Alvaro’s slide regarding how IMS should be implemented, and the fact that the selectors are not independent, despite their name. As he noted, these requirements have not been implemented consistently in current ECDIS systems which points to ambiguity/gaps in the requirements and/or test procedures. Despite this, the S-101 portrayal catalogue implementation matches what you describe, with some minor additions as noted below. 

As to your concerns regarding changes, I don’t feel they are warranted: 

![image](https://github.com/S-101-Portrayal-subWG/Working-Documents/assets/70189016/3afa492e-cd1f-4e9e-8f24-a57748370ace)

**Comment by alvarosanuy** [comment]
@DavidGrant-NIWC - Based on the discussions at S101PT and your exchange with Hannu, **are you planning any S-98 change proposal to further clarify IMS implementation guidance? Do you still see the need for this?**

Please note that I added a new label to highlight 'that a 'new (or updated) S-164 test' is required. For this, S-98 guidance and statements must be finalised first.

**Comment by DavidGrant-NIWC** [comment]
The implementation matches the requirements as written, so we don't plan to develop a change proposal. Our focus is on ensuring that S-100 / S-101 can meet requirements. I agree that more detailed S-164 tests should be provided to remove ambiguity.

I do think simplifying the functionality would benefit everyone, but a change proposal would have a better chance of acceptance if it came from an OEM or industry group.

**Comment by alvarosanuy** [comment]
@DavidGrant-NIWC - Refer to S101PT11 Action below:

![image](https://github.com/S-101-Portrayal-subWG/Working-Documents/assets/70189016/88570988-19d3-4de5-8bac-a08ff824e485)

Would you mind working with @rmalyankar & @kusala9 to ensure the following key functional requirements are included in S-98 and related S-164 tests developed, as required, to check compliance?
Please let me know if you think my interpretation is incorrect or incomplete.

- Selecting a Display Category must turn ON all VG/VGL allocated to that Display Mode in S-98 with the exception of IMS Objects.
- Objects allocated to a 'higher' Display category (i.e Other) can be added to a lower category (i.e. Standard) by individually activating their associated VGL. This action must not load all other layers allocated to their default Display Category (i.e. Turning ON 'Accuracy' when in Standard Display must not switch the display mode to 'Other').
- When additional VG/VGL are manually added by the user to a Display category, ECDIS should add the character '+' at the end of the name of the Display Mode (i.e. Standard+ when Accuracy is added to the Standard Display category).
- If a VG/VGL is active, IMS related to the objects in that VG/VGL can be activated by the Mariner without triggering the loading of all VGL related to the Display category the IMS relates to [i.e. if LIGHTS have been manually added to Base Display (Base+)], 'Date Dependent' and 'Highlight Date Dependent' symbols must be shown if they apply to LIGHTS).
- Once a Display Mode is selected by the Mariner, it shouldn't be possible to turn 'core' VG/VGL OFF. Mariners should load a lower Display category first (i.e. Base) and then add additional layers as required to build their preferred view.

**Comment by HolgerBothien** [comment]
Agree with most of the above.
Except the last point. It is not a good idea to force the user to do things in a special order. (doing things first in order to allow other things). I am even not sure how far we should go with the specification. I would leave room for clever solutions by OEMs as long as some general rules are covered.

Just one example of how VG/VGL could be handled:
1. Having a GUI element to choose 
 - Displaybase
 - Standard
 - Other
 - Custom
2. when choosing Custom some GUI element will be enabled to select the VGLs similar to 

![image](https://github.com/S-101-Portrayal-subWG/Working-Documents/assets/88826517/84aa30b9-771e-4607-99ad-902122783a40)

This an approach that is easy to understand, either one of the three predefined modes or my own choice.
IMS are as the name says independent of this with own GUI elements and do not have any connections to VG/VGL even some are using viewing groups. The user does not have knowledge about the implementation of an IMS. It must just work, i.e. switching ContourLabels on shows the labels on all contours that are visible **independent** of what viewing groups are currently selected. Thus, they may only appear on the safety contour if the viewing group for contours isn't selected.

**Comment by DavidGrant-NIWC** [comment]
@alvarosanuy, we are happy to work with Jonathan on improving the S-164 tests for IMS.

WRT S-98, I agree with @HolgerBothien that OEM's should be provided some leeway in implementing their UI's. I doubt that they will want to change their current implementations. I also don't think your proposed changes would impact the current S-101 IMS implementation (the IMS would still be dependent on the display mode).

It seems like the following so-called "Independent Mariner Selectors" should just be viewing group layers. We could consider the following changes which better reflect those IMS which are dependent on the selected display mode. It does involve making some minor changes to the S-52 viewing group layer requirements, for instance moving VG `33021` and `33022` from `Miscellaneous (Other)` to `Depth contours` and `Contour Labels`.

|IMS|Change from current implementation|
|:---|:---|
|Accuracy|Make it a VGL; a sub-element of the `Miscellaneous (Other)` VGL. That way it's obvious that it's part of the `Other` display mode|
|Contour Label|Make it a VGL; a sub-element of the `Depth contours` VGL. Part of the `Other` display mode|
|Highlight Info|Make it a VGL; a sub-element of the `Miscellaneous (Other)` VGL. Part of the `Other` display mode|
|Highlight date dependent|Same as highlight info|
|Highlight document|Same as highlight info|
|Shallow Pattern|Coordinate with ENCWG to make independent of display mode. Retain current independent selector. Alternatively, make it a VGL; part of the `Standard` display mode.|
|Unknown|Make it a VGL; part of the `Standard` display mode.|
|Update review|Retain current "independent" selection.|

![image](https://github.com/S-101-Portrayal-subWG/Working-Documents/assets/62906211/2edfc75b-4ab1-4dde-b38a-52383adf2f6f)

**Comment by alvarosanuy** [comment]
@HolgerBothien & @DavidGrant-NIWC - Thanks for your input and agree with your comments. Happy to accommodate changes to VG/VGL as required to provide a functionality (and terminology) that better aligns with how things work in practice.

It may be a good time to squeeze these small changes into a new version of S-52, as it should be developed to accommodate latest IMO requirement on the use of Vert/Hor Uncertainties in ECDIS, anyway. If not possible, it would be up to OEMs to tie up VG/VGL and functionality in DF-ECDIS ...

@kusala9 - Can you coordinate with Tom Mellor regarding proposed IMS changes to S-98 and their impact on S-52?

**Comment by DavidGrant-NIWC** [comment]
Based on the discussions during the meeting today, we recommend that the following changes are made in coordination with the ENCWG:

Update the documents to indicate that the following viewing groups are independent of the selected display mode. The indicated viewing group layer should be updated to remove the viewing group.
|VG|Symbol|Remove from viewing group layer|
|:---:|:---:|:---:|
|`31010`|`CATZOC`|`18`
|`31011`|`LOWACC01`|`18`
|`31030`|`INFORM01` (Highlight Info)|`18`|
|`31031`|`INFORM01` (Highlight Doc)|`18`|
|`31032`|`CHDATD01`|`18`|
|`23010`|Shallow water pattern (diamonds)|`10`|
|`21010`|`QUESMRK1` (unknown object)|`3.1`|
|`33021`|Safety contour label|`18`|
|`33022`|Contour label (other than the safety contour)|`18`|

The following viewing group layers are added to implement the indicated IMS. The viewing group layers are independent of the selected display mode.

|IMS|Viewing group layer|Viewing groups|
|:---|:---:|:---|
|`Accuracy`|TBD|`31010`, `31011`|
|`Feature Highlights`|TBD|`31030`, `31031`, `31032`|
|`Highlight info`|TBD - sub-element of `Feature Highlights`|`31030`|
|`Highlight document`|TBD - sub-element of `Feature Highlights`|`31031`|
|`Highlight date dependent`|TBD - sub-element of `Feature Highlights`|`31032`|
|`Shallow pattern`|TBD|`23010`|
|`Unknown`|TBD|`21010`|
|`Update review`|TBD|Recommend that SY(CHRVDELN) etc. are assigned a new, fixed viewing group rather than inheriting the vg from the updated symbol.|
|`Contour label`|TBD|`33021`, `33022`|

**Comment by kusala9** [comment]
reference https://github.com/iho-ohi/98-interoperability/issues/27 (the S-98 issue covering this) can we propose the S-98 modifications required and I'm happy to draft the changes in the redline. I'll talk to Tom about the S-52 issue as well.

**Comment by alvarosanuy** [comment]
**Decisions made at Portrayal subWG meeting on 18/10/23**

- IMS functional requirements and the corresponding changes to VG/VGL to make them effective were agreed using @DavidGrant-NIWC proposal above as an starting point. https://github.com/S-101-Portrayal-subWG/Working-Documents/issues/18#issuecomment-1767117399
- The PsWG approved the removal of certain VGs from VGL as detailed in the **top table** included in @DavidGrant-NIWC comment referred above.
- In regards to the **bottom table**, it was decided that all listed VGs should be allocated a new number (90000) and that each IMS will be allocated a new and unique VGL (TBD)

**Accordingly the following Actions are required:**

- [ ] **NIWC**- Upload a new version of the 'bottom' table based on the decisions made above. Please propose new VGL numbers for each IMS.
- [ ] @kusala9 & @DavidGrant-NIWC  - Coordinate the submission of the corresponding S-98 change proposals to S100WG8.

This issue will continue open until feedback is provided on the outcome of the change proposals (post S100WG8 meeting).

If approved by S1001WG8:

- [ ] @kusala9 - Develop new S-164 Test to check ECDIS implementation compliance.

**Comment by DavidGrant-NIWC** [comment]
https://github.com/S-101-Portrayal-subWG/Working-Documents/issues/123#issuecomment-1786357751 requested removal of `Unknown` from the IMS, This change would require that `21010` is added to either a new or existing VGL (because otherwise it will not be part of any display mode). Our recommendation is that `21010` is added to VGL 10 - `Miscellaneous (Standard)` and VGL 10a - `Chart (Standard)`:
![image](https://github.com/S-101-Portrayal-subWG/Working-Documents/assets/62906211/3ce0b9f9-5ae1-40fa-9513-9d1d2b3b2e2d)

**Comment by alvarosanuy** [comment]
> [#123 (comment)](https://github.com/S-101-Portrayal-subWG/Working-Documents/issues/123#issuecomment-1786357751) requested removal of `Unknown` from the IMS, This change would require that `21010` is added to either a new or existing VGL (because otherwise it will not be part of any display mode). Our recommendation is that `21010` is added to VGL 10 - `Miscellaneous (Standard)` and VGL 10a - `Chart (Standard)`: 

**Supported. 'Unknown feature' should not be considered an IMS (refer to #123). Strongly recommend it remains in 'Standard' DM.
Endorse the proposal to remove VG 21010 from VGL 3.1 and move it to VGL10 & 10a.**

**Comment by alvarosanuy** [comment]
@DavidGrant-NIWC - **can you confirm PC has been updated and clarify how 'Unknown feature' would be handled?**. See my comment above.
 If pending tasks are S-98 only then, close and transfer issue to the corresponding GitHub space.

**Comment by DavidGrant-NIWC** [comment]
> [...] clarify how 'Unknown feature' would be handled?

See row 1 below; it's part of the Standard display using viewing group 21010. There is no longer an IMS to toggle the display of unknown objects.

I suggest closing this issue if you agree these changes satisfy the request; any additional changes can be handled as new issues targeting the specific requested change. An issue has been entered in S-98 to address these changes: [S-98 #38](https://github.com/iho-ohi/98-interoperability/issues/38). An S-164 has also been entered to address any changes which might be required to the test procedures: [S-164 #75](https://github.com/iho-ohi/S-164-Sub-Group/issues/75).

### Re-mapping of viewing groups controlled by IMS
|S-52 VG|Symbol|Remove from VGL|S-101 VG|IMS VGL|Notes
|:---:|:---|:---:|:---:|:---:|:---
|`21010`|`QUESMRK1` (unknown object)|`3.1`|N/A|N/A|Implemented via https://github.com/iho-ohi/S-101_Portrayal-Catalogue/issues/320, no longer an IMS
|`23010`|Shallow water pattern (diamonds)|`10`, `10a`|`90000`|`900`
|`31010`|`CATZOC`|(see note)|`90010`|`910`|`31010` is retained for `QoS`, `QoNBD`, `SoundingDatum`, `VerticalDatumOfData`; CATZOC pattern toggles via `90010` or IMS `910`
|`31011`|`LOWACC01`|`18`|`90011`|`910`
|`31030`|`INFORM01` (Highlight Info)|`18`|`90020`|`920`,`920a`
|`31031`|`INFORM01` (Highlight Doc)|`18`|`90021`|`920`,`920b`
|`31032`|`CHDATD01` (Highlight Date Dependent)|`18`|`90022`|`920`,`920c`
|`33021`|Safety contour label|`18`|`90030`|`930`
|`33022`|Contour label (other than the safety contour)|`18`|`90031`|`930`
|Inherited|Update review|N/A|`90040`|`940`|UpdateInformation symbols. S-52 inherited VG from updated object; S-101 always uses `90040`.

### Viewing group layers and viewing groups for each IMS (duplicates info from the previous table)
|IMS|Viewing group layer|Viewing groups|
|:---|:---:|:---|
|`Unknown`<sup>1</sup>|`10`, `10a`|`21010`|
|`Shallow pattern`|`900`|`90000`|
|`Accuracy`|`910`|`90010`, `90011`|
|`Feature Highlights`|`920`|`90020`, `90021`, `90022`|
|`Highlight info`|`920a` (sub-element of `Feature Highlights`)|`90020`|
|`Highlight document`|`920b` (sub-element of `Feature Highlights`)|`90021`|
|`Highlight date dependent`|`920c` (sub-element of `Feature Highlights`)|`90022`|
|`Contour labels`|`930`|`90030`, `90031`|
|`Update review`|`940`|`90040`|

<sup>1</sup> - `Unknown` is no longer an IMS. It is part of the STANDARD display mode.

**Comment by alvarosanuy** [comment]
Implemented in PC 1.2.0
